cg_lims/status_db_api.py

```diff
@@ -1,20 +1,34 @@
-import json
-
 import requests
+from typing import List
+from urllib.parse import urljoin
 
 from cg_lims.exceptions import LimsError
+from cg_lims.models.sequencing_metrics import SequencingMetrics
 
 
-class StatusDBAPI(object):
-    def __init__(self, url=None):
-        self.url = url
+class StatusDBAPI:
+    def __init__(self, base_url):
+        self.base_url = base_url
 
-    def apptag(self, tag_name, key=None, entry_point="/applications"):
+    def _get(self, endpoint, key=None):
+        url = urljoin(self.base_url, endpoint)
         try:
-            res = requests.get(self.url + entry_point + "/" + tag_name)
+            response = requests.get(url)
+            response.raise_for_status()
+            data = response.json()
             if key:
-                return json.loads(res.text)[key]
-            else:
-                return json.loads(res.text)
-        except ConnectionError:
-            raise LimsError(message="No connection to clinical-api!")
+                return data.get(key)
+            return data
+        except requests.RequestException as e:
+            raise LimsError(f"Failed to get data from {url}, {e}")
+
+    def apptag(self, tag_name, key=None):
```

Comment:
likely a pre-existing function and out of the scope. But just in case it would be an easy fix
```suggestion
    def get_application_tag(self, tag_name, key=None):
```

Reply:
Created a separate PR for it https://github.com/Clinical-Genomics/cg_lims/pull/410